server/pulp/server/managers/content/upload.py

```diff
@@ -180,8 +181,12 @@ def import_uploaded_unit(repo_id, unit_type_id, unit_key, unit_metadata, upload_
         :type  unit_metadata: dict
         :param upload_id:     upload being imported
         :type  upload_id:     str
-        :return:              A SyncReport indicating the success or failure of the upload
-        :rtype:               pulp.plugins.model.SyncReport
+        :return:              A dictionary describing the success or failure of the upload. It must
+                              contain the following keys:
+                                'success_flag': bool. Indicates whether the upload was successful
+                                'summary':      json-serializable object, providing summary
+                                'details':      json-serializable object, providing details
+        :rtype:               dict
```

Comment:
A couple of ``:raises`` entries would be helpful, both for the ``PulpCodedException`` you added, and the existing ``MissingResource`` while you're at it.